src/audio/mic_privacy_manager/mic_privacy_manager_intel.c

```diff
@@ -0,0 +1,232 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2025 Intel Corporation. All rights reserved.
+
+#include "sof/audio/mic_privacy_manager.h"
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/mic_privacy/intel/mic_privacy.h>
+#include <zephyr/logging/log.h>
+#include <sof/audio/audio_stream.h>
+#include <sof/audio/buffer.h>
+#include <sof/audio/mic_privacy_manager.h>
+
+const struct device *mic_priv_dev;
+
+static struct mic_privacy_api_funcs *mic_privacy_api;
+static enum mic_privacy_policy mic_privacy_policy;
+
+#define LOG_DOMAIN mic_priv
+
+LOG_MODULE_REGISTER(LOG_DOMAIN);
+
+void handle_dmic_irq(void *const self, int a, int b)
+{
+	LOG_DBG("mic_privacy handle_dmic_irq");
+
+	if (mic_privacy_api->get_dmic_irq_status()) {
+		uint32_t mic_disable_status = mic_privacy_api->get_dmic_mic_disable_status();
+		struct mic_privacy_settings settings;
+
+		mic_privacy_fill_settings(&settings, mic_disable_status);
+		mic_privacy_propagate_settings(&settings);
+		mic_privacy_api->clear_dmic_irq_status();
+	}
+}
+
+void handle_fw_managed_irq(void * const dev)
```

Comment:
is this `void * const` type deliberate? You do mean, that the variable cannot be changed, while the data it points to can be changed? This is less common and probably less useful, but that's really what you want here, that's ok too

Reply:
fixed, inside zephyr driver this is expected form: void (*routine)(const void *parameter)